### PR TITLE
[cmake] support setting log level

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: script/check-simulation-build-cmake [hunk 1]
```diff
@@ -41,6 +41,8 @@ build_all_features()
     reset_source
     "$(dirname "$0")"/cmake-build simulation
     reset_source
+    "$(dirname "$0")"/cmake-build simulation -DOT_LOG_LEVEL=INFO
+    reset_source
```
Comment (simonlingoogle) [inline script/check-simulation-build-cmake:45]: `LOG_LEVEL=INFO` build test seems not that necessary? Note that each test we add makes our GitHub action slower. 

I think it's better to have LOG_LEVEL=NONE test and LOG_LEVEL=DEBG test to cover most cases (not sure if we already have them).

Comment (bukepo) [reply script/check-simulation-build-cmake:45]: I think this only checks building.

Comment (simonlingoogle) [reply script/check-simulation-build-cmake:45]: Currently, build tests are the slowest github actions we have. 
That's why I think we should add most representative building tests (i.e. full log building and none log building), but need not to add intermediate building tests. 

Thoughs?

Comment (bukepo) [reply script/check-simulation-build-cmake:45]: The new test is only for validating the cmake option OT_LOG_LEVEL works, nothing beyond this purpose.

**Review by simonlingoogle — APPROVED** [review]
lgtm, thanks

**Pull request merged**